### ISSUE TITLE
NX3000

### ISSUE THREAD

**Issue by JLuc** [ge0rg/samsung-nx-hacks#1]
Nx3000 is described as compatible.

Could you provide tools for nx3000 ?
Or are there tools for nx3000 ? 
I'm on Ubuntu and basically i'd like to access its SD card via wifi.

**Comment by DougHickok** [comment]
Try the "FTP Access Point Script" from here:
https://sites.google.com/site/nxcryptophotography/other-scripts
I'm curious how similar the camera software is between the NX300 and NX3000.  Let me know if this works, or what errors you're seeing in the log file (on the card after it runs), or if no log file appears.

**Comment by anonym-snegl** [comment]
Hello! Was wondering if you had luck running the scripts on an NX3000. I've been trying to use the remote viewfinder to get a livestream (described here: https://github.com/ge0rg/samsung-nx-hacks/wiki/Connecting-as-remote-viewfinder-to-NX-camera) but I keep getting a HTTP/1.1 503 error. 

Anybody else have any experience getting a livestream from an Nx3000? Could the error be that the request is sent as HTTP/1.0 but should be 1.1?

Any info appreciated!

**Comment by Nikolas-LFDesigns** [comment]
Sorry for not being "anybody else" (as article author), but I might help as well )
What command sequence are you trying? Did you get on-camera pairing confirmation? And yes, you should persuit the exact command or else your client would be simply rejected.

**Comment by GithubUserMathias** [comment]
Hello,
 
I don't know the NX3000 but I have been able to control the SX520HS camera from canon by a program I've written in C#.
The livestreem I've got from the SX520HS is a low quality one. I'm not sure it is sending the High quality stream.
 
It's possible to send commands for taking a phote, controlling the zoom and transmitting the photo to the PC for the SX520HS.
If you want to know how it works you can use wireshark to listen to the data. I've installed wireshark on a phone and used it to get the commands which are send and received by the 'Canon Camera Connect' app.
 
Kind Regards,
Matthijs

 
Date: Sun, 22 Nov 2015 13:17:55 -0800

**Comment by Nikolas-LFDesigns** [comment]
Curious how did you get a wireshark connection on phone? Tried it in past but it even didn't want to start anything, let alone sniff packets or at least establish a proxy..

**Comment by fuctface** [comment]
Try Kali Linux NetHunter

**Comment by PatrikSamuelTauchim** [comment]
You can setup wireless bridge on your PC (between camera and phone) and run Wireshark on it.

If you just cannot get you PC paired with camera, because it is not sending pairing request, take MAC address from PC and change MAC of your phone to this address. Then pair camera with phone normally and camera will have this address on the list of paired devices. Then change MAC address of your phone back and connect with PC, which is now paired.

Personally, I'm not interested in this kind of controlling camera anymore. Stream is low quality and it is also not possible to shoot to .RAW  while using remote control. Do not need to mention effect on the battery life. Not sure how it's for new cameras, but my NX1000 is not capable of USB charging or power, so I'm limited with battery life, even while home.

For remote shutter feature I'm working on Bluetooth remote shutter instead.

**Comment by fuctface** [comment]
Hey PatrikSamuelTauchim, I also have an nx1000. Thats why I am following this project. Are you trying to develop remote shutter for that camera? Have you been able to use any of the info here with nx1000? I started mess around with it a few months ago, ran BinWalk tool on the firmware and figured out it was VxWorks based and kinda lost interest. Have you been able to do anything interesting? I didnt know it was bluetooth capable.

**Comment by PatrikSamuelTauchim** [comment]
Hello @fuctface , I have working prototype of Bluetooth shutter (dedicated HW connected into camera microUSB) and somehow working Android app (not published yet).
More info at:
https://github.com/PatrikSamuelTauchim/NX-Bluetooth-Remote

**Comment by GithubUserMathias** [comment]
Hello Nikolas,

To use wireshark on a phone you have to root it. I don't know too much about that so I used an old phone and a tutorial from the internet. Then with the app 'Shark for Root' I could run a scan on the network adapter. This app creates a file which your PC version of Wireshark can read.

Kind regards,
Matthijs

**Comment by ge0rg** [comment]
I've obtained a semi-broken NX3000 and disassembled it. The processor is a Fujitsu / Socionext Milbeaut MB86S22A, and it is not running Linux. It's an ARM v7 CPU, but we have no source code from Samsung, no CPU specifications and not even a clue of how the firmware is encoded/compressed, so changing anything in there will be very hard.

I've documented the hardware at https://github.com/ge0rg/samsung-nx-hacks/wiki/NX3000-Hardware-Information
